app/models/solidus_paypal_commerce_platform/paypal_order.rb

```diff
@@ -44,10 +44,16 @@ def payer
     end
 
     def name(address)
-      {
-        given_name: address.firstname,
-        surname: address.lastname
-      }
+      if greater_than_2_10
+        {
+          given_name: address.name,
+        }
+      else
+        {
+          given_name: address.firstname,
+          surname: address.lastname
+        }
+      end
```

Comment:
Are we sure PayPal prefers the given_name over the surname?
Might be useful to have a link to the reference documentation in a comment.

Reply:
Surname doesn't look required on [PayPal documentation ](https://developer.paypal.com/docs/api/orders/v2/#definition-name), but to keep the same struct I have added it back.